tsconfig.json

```diff
@@ -0,0 +1,18 @@
+{
+  "include": ["src/**/*.ts"],
+  "compilerOptions": {
+    "target": "ES2022",
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
+
+    // https://esbuild.github.io/content-types/#typescript
+    "esModuleInterop": true,
+    "isolatedModules": true,
+    "strict": true,
+
+    // affects `tsc` only
+    "declaration": true,
+    "emitDeclarationOnly": true,
+    "outDir": "dist"
+  }
+}
```

Comment:
I am not sure if we've copy&pasted this from somewhere else, but in case you are curious about it, I usually take [this cheat sheet](https://www.totaltypescript.com/tsconfig-cheat-sheet) as the starting point for every TS Project. 

I'm not saying we should follow it for this particular case, but maybe there's something interesting to be added to this `tsconfig.json` file, e.g.

* `"noUncheckedIndexedAccess": true`: By default, when you access an array or object with an index, TypeScript assumes the value will always exist, however, when this option is enabled, TypeScript will treat such accesses as potentially undefined and force you to handle the case where the value might not exist

Reply:
That's a great cheat sheet! I like the idea of having opinionated defaults here for CARTO stuff. We'll need to be a little careful, in this specific case, as ESBuild does not implement all `tsconfig.json` options, I've just enabled strict:true and noUncheckedIndexedAccess:true for now.